hammett/core/bot.py

```diff
@@ -123,13 +123,17 @@ def _get_handler_object(
             UnknownHandlerType: If the handler type is unknown.
 
         """
+        from hammett.widgets.calendar_widget import DetailedTelegramCalendar
+
         handler_object: CallbackQueryHandler[Any, Any] | MessageHandler[Any, Any]
-        if handler_type in {HandlerType.BUTTON_HANDLER, ''}:
+        if handler_type in {HandlerType.BUTTON_HANDLER, HandlerType.CALENDAR_HANDLER, ''}:
             handler_object = CallbackQueryHandler(
                 handler,
                 # Specify a pattern. The pattern is used to determine which handler
                 # should be triggered when a specific button is pressed.
-                pattern=calc_checksum(handler),
+                pattern=DetailedTelegramCalendar.func() if
+                handler_type == HandlerType.CALENDAR_HANDLER
+                else calc_checksum(handler),
```

Comment:
There's a formatting problem. Maybe something like this?
```suggestion
                pattern=(
                    DetailedTelegramCalendar.func() 
                    if handler_type == HandlerType.CALENDAR_HANDLER
                    else calc_checksum(handler)
                ),
```